rust/noosphere/src/ffi/tracing.rs

```diff
@@ -0,0 +1,65 @@
+use noosphere_core::tracing::{initialize_tracing, NoosphereLog};
+use safer_ffi::prelude::*;
+
+const NOOSPHERE_LOG_SILENT: u32 = 0;
+const NOOSPHERE_LOG_BASIC: u32 = 1;
+const NOOSPHERE_LOG_CHATTY: u32 = 2;
+const NOOSPHERE_LOG_INFORMED: u32 = 3;
+const NOOSPHERE_LOG_ACADEMIC: u32 = 4;
+const NOOSPHERE_LOG_TIRESOME: u32 = 5;
+const NOOSPHERE_LOG_DEAFENING: u32 = 6;
+
+#[derive_ReprC(rename = "ns_noosphere_log")]
+#[repr(u32)]
+#[ffi_export]
+/// Configuration presets for Noosphere log behavior
+pub enum NsNoosphereLog {
+    /// Equivalent to minimal format / `OFF` filter
+    Silent = NOOSPHERE_LOG_SILENT,
+    /// Equivalent to minimal format / `INFO` filter
+    Basic = NOOSPHERE_LOG_BASIC,
+    /// Equivalent to minimal format / `DEBUG` filter
+    Chatty = NOOSPHERE_LOG_CHATTY,
+    /// Equivalent to verbose format / `DEBUG` filter
+    Informed = NOOSPHERE_LOG_INFORMED,
+    /// Equivalent to pretty format / `DEBUG` filter
+    Academic = NOOSPHERE_LOG_ACADEMIC,
+    /// Equivalent to verbose format / `TRACE` filter
+    Tiresome = NOOSPHERE_LOG_TIRESOME,
+    /// Equivalent to pretty format / `TRACE` filter
+    Deafening = NOOSPHERE_LOG_DEAFENING,
```

Comment:
While examples should this working as expected, enums exported via safer_ffi for error constants did not put a `_` between enum name and value -- does header generate `NS_NOOSPHERE_LOG_SILENT` or `NS_NOOSPHERE_LOGSILENT`?

also, consider: `ns_noosphere`/`NsNoosphere` refers to the top-level context object, which logs can be thought of as related. Similarly, maybe `ns_log` (e.g. `NS_LOG_SILENT`) is sufficient (though may be fighting some Apple **N**ame **S**paces), though not unique to this API

Reply:
The related enum is literally `NoosphereLog`, and is a counterpart to the accepted values of the `NOOSPHERE_LOG` environment variable. `NsNoosphere` is the top-level context object, indeed.

I would maybe consider adding a prefix to the constants `NS_NOOSPHERE_LOG_SILENT`. But, to my eyes truncating e.g., `NOOSPHERE_LOG_*` to `NS_LOG_*` is moving away from legibility because it means we specify the log level with different language in the C FFI compared to both the Rust API and the environment variable.